src/rewrite/vct/rewrite/Disambiguate.scala

```diff
@@ -105,13 +124,22 @@ case class Disambiguate[Pre <: Generation]() extends Rewriter[Pre] {
         else if(op.isMapOp) MapGet(dispatch(collection), dispatch(index))(op.blame)
         else if(op.isArrayOp) ArraySubscript(dispatch(collection), dispatch(index))(op.blame)
         else if(op.isSeqOp) SeqSubscript(dispatch(collection), dispatch(index))(op.blame)
+        else if(op.isVectorOp) VectorSubscript(dispatch(collection), dispatch(index))(op.blame)
         else throw Unreachable("AmbiguousSubscript must subscript a pointer, map, array, or seq because of the type check.")
       case op @ AmbiguousMember(x, xs) =>
         if(op.isMapOp) MapMember(dispatch(x), dispatch(xs))
         else if(op.isSetOp) SetMember(dispatch(x), dispatch(xs))
         else if(op.isBagOp) BagMemberCount(dispatch(x), dispatch(xs))
         else if(op.isSeqOp) SeqMember(dispatch(x), dispatch(xs))
         else throw Unreachable("AmbiguousMember must query a map, set, bag, or seq because of the type check.")
+      case cmp: AmbiguousComparison[Pre] =>
+        if(cmp.isVectorOp) cmp match {
+          case AmbiguousEq(left, right, _) => VectorEq(dispatch(left), dispatch(right))
+          case AmbiguousNeq(left, right, _) => VectorNeq(dispatch(left), dispatch(right))
+        } else cmp match {
+          case AmbiguousEq(left, right, _) => Eq(dispatch(left), dispatch(right))
+          case AmbiguousNeq(left, right, _) => Neq(dispatch(left), dispatch(right))
+        }
```

Comment:
Is this vector inner type in `AmbiguousEq` used for something still? It seems like a strange abstraction.

Reply:
See the comment I made below